merian.md first setup (#51)
eurec4a/how_to_eurec4a: how_to_eurec4a/merian_cloudradar.md

---
jupytext:
  text_representation:
    extension: .md
    format_name: myst
    format_version: 0.13
    jupytext_version: 1.11.1
kernelspec:
  display_name: Python 3
  language: python
  name: python3
---

# Cloud radar data collected on MS Merian

+++

## General information

> From here you can access the W band radar data collected on the RV MSMerian. The radar is a frequency modulated continuous-wave (FMCW) 94 GHz dual polarization radar (manufactured by RPG GmbH) and the data are published on AERIS. The data presented here are corrected for ship motions. For more details on the algorithm applied for correction, please check the paper in preparation in ESSD.

Author: yes, I also would love to mention that this tool is available in the data paper, can i?

> From here you can access the W band radar data collected on the RV MSMerian. The radar is a frequency modulated continuous-wave (FMCW) 94 GHz dual polarization radar (manufactured by RPG GmbH) and the data are published on AERIS. The data presented here are corrected for ship motions. For more details on the algorithm applied for correction, please check the paper in preparation in ESSD.

Reviewer: We've recently enabled citations via bibtex (#54) within the book. The citation style should still be adjusted (#55), but that's independent of using the feature.

We are still working on how to best refer to the book from outside (#53). Currently, there's a handle-link to the book (http://doi.org/21.11101/0000-0007-EBF5-C), which works, but has the disadvantage that it doesn't allow pointing to individual articles and is hard to remember. We're working towards using `https://howto.eurec4a.eu`, but there are currently some admisitrative barriers which should be resolved soon. See #53 for details / updates on that matter.

If you still can wait a few days before placing references into the book, I'd recommend to wait until these things have settled.

It measured continously during the campaign, with a time resolution of 3 seconds and vertical range resolution varying with height between 7.5, 9.2 and 30 m. Here we provide you with some tools to plot and visualize the data.   
A **quicklook browser** (https://bit.ly/2QpsAzh) is helping you to pick the case you are more interested in. If you have questions or if you would like to use the data for a publication, please don't hesitate to get in contact with the dataset author that you can find here (https://doi.org/10.25326/156).

+++

### Getting the data catalog

The python code below allows you to browse through the days avaiable

```{code-cell} ipython3
import intake
from datetime import datetime
import matplotlib.pyplot as plt
import numpy as np

path_to_catalog = 'https://raw.githubusercontent.com/eurec4a/eurec4a-intake/master/catalog.yml'
cat = intake.open_catalog(path_to_catalog)
```

### Getting the data from the MS-Merian ship
To visualize which datasets are available for the MS-Merian ship, type:

```{code-cell} ipython3

> print('List of data/instruments available from Merian:')

Author: done

print(list(cat['MS-Merian']))
```

### Check available days of Wband radar data
To check which days are available from the Wband radar dataset, type:

```{code-cell} ipython3
print('W band dataset:')
for key, source in cat['MS-Merian']['FMCW94_RPG'].items():
    desc = source.describe()
    user_parameters = desc.get("user_parameters", [])
    if len(user_parameters) > 0:
        params = " (" + ", ".join(p["name"] for p in user_parameters) + ")"
    else:
        params = ""
    print(f"{key}{params}: {desc['description']}")
    for parameter in user_parameters:
        print(f"    {parameter['name']}: {parameter['min']} ... {parameter['max']} default: {parameter['default']}")
```

### Selecting one day for plot/visualization

> To work on a specific day, please provide the date as input in the form "yyyy-mm-dd hh:mm" as a field to date in the code below

Author: Hey Tobi.. sorry, I don't get what you exactly mean here... I checked the link briefly but I am still not getting what options you are talking about, can you make me an example? thank you!

> To work on a specific day, please provide the date as input in the form "yyyy-mm-dd hh:mm" as a field to date in the code below

Reviewer: Sure. The thing is, `intake` converts the given datetime information automatically into what is needed by the catalog from multiple different sources. If you take the date and time from a calendar, via email, by phone etc..., than you'll likely have it in form of a string, at least it should be simple to format in an [ISO8601](https://en.wikipedia.org/wiki/ISO_8601) compliant way. Thus it is convenient to use:
```python
date = '2020-01-27 13:00'
```
It may however also happen, that you got the time information from another script, which provides the date as [standard Python `datetime`](https://docs.python.org/3/library/datetime.html) object, in that case, the following would be more convenient:
```python
date = datetime.datetime(2020, 1, 27, 13). # of course this datetime would come from another function, dataset, etc...
```
by the way, `intake` works also perfectly fine with [`numpy.datetime64`](https://numpy.org/doc/stable/reference/arrays.datetime.html) objects. So say that you know that there was some interesting event on `2020-01-26T11:00`, but you like to know what happened to the radar 26 hours after the fact, you might have
```python
event = np.datetime64("2020-01-26T11:00"). # this event again might come from some other code
date = event + np.timedelta64(26, 'h')
```
and if you are more old fashioned, then your scripts might still use [UNIX timestamps](https://en.wikipedia.org/wiki/Unix_time), which is also covered by intake:
```python
date = 1580130000
```

---

I don't think that you'll have to enumerate all those options in particular. But it might be a useful hint to some people that you don't have to convert to this string representation of the date if you already have your reference date and time in other formats, i.e. this would **not** be a good idea:
```python
event = np.datetime64("2020-01-26T11:00")  # comes from somewhere else...
temp = str(event + np.timedelta64(26, 'h'))
day = temp[:10]
time = temp[11:]
date = day + " " + time
```

> To work on a specific day, please provide the date as input in the form "yyyy-mm-dd hh:mm" as a field to date in the code below

Author: I just added the line: "Note: the datetime type accepts multiple values: Python datetime, ISO8601 string, Unix timestamp int, “now” and “today”.

```{code-cell} ipython3
print('select the date and read the dataset')
date = '2020-01-27 13:00' # <--- provide the date here 
ds = cat['MS-Merian']['FMCW94_RPG'].motion_corrected(date=date).to_dask()
print(ds)
```

### Plot some radar quantities
To create time/height plots for radar moments, pick a variable name having dimension (time, height) from the list of data variables and type the name in the code below. The example here is for mean Doppler velocity (corrected for ship motions). You can decide to either select the entire day or a given hour by providing time_min and time_max.
To provide time_min and time_max, modify the string 'yyyy-mm-ddThh:mm:ss' in red. <br> 
Example: <br>
to select the entire day :<br>
    - time_min = np.datetime64('2020-01-27T00:00:00') <br>
    - time_max = np.datetime64('2020-01-27T23:59:59')<br>
to select between 13:00 and 15:00 UTC:<br>
    - time_min = np.datetime64('2020-01-27T13:00:00') <br>
    - time_max = np.datetime64('2020-01-27T15:00:00')<br>

```{code-cell} ipython3
# set min and max time values for plotting along the x-axis
time_min = np.datetime64('2020-01-27T13:00:01') # insert the string value corresponding to t_min

> time_max = np.datetime64('2020-01-27T15:00:01') # insert the string value corresponding to t_max

Reviewer: I am wondering why these times are off by one second as compared to the example?

Author: there' s no specific  reason. we can set 00

# selecting subset of data
ds_sliced = ds.sel(time=slice(time_min, time_max))
```

```{code-cell} ipython3
fig, ax = plt.subplots(figsize=(12, 6))
# set here the variable from ds to plot, its color map and its min and max values
ds_sliced.mean_doppler_velocity.plot(x='time', y='height', cmap="seismic", vmin=-10., vmax=10.)

> plt.ylim(0, 4500)
> ax.spines['right'].set_visible(False)
> ax.spines['top'].set_visible(False)
> ax.set_title("Mean doppler velocity for : "+date)
> ax.set_xlim(time_min, time_max)

Reviewer: I recommend to use `ax.set_ylim` to make it consistent.
```suggestion
ax.spines['right'].set_visible(False)
ax.spines['top'].set_visible(False)
ax.set_title("Mean doppler velocity for : "+date)
ax.set_xlim(time_min, time_max)
ax.set_ylim(0, 4500)
```

#plt.xlim('2020-01-19 20:00', '2020-01-19 21:00')
```

### Check corresponding merian position in the selected hour

> We use the function track2layer defined in the Interactive HALO tracks chapter, and we provide as input the data for the time interval selected during the day

Reviewer: I'd recommend to add backticks for the code and an internal link directly to the referenced chapter.
```suggestion
We use the function `track2layer` defined in the {doc}`flight_tracks_leaflet` chapter, and we provide as input the data for the time interval selected during the day
```

```{code-cell} ipython3
import ipyleaflet

# function to associate features to a given track
def track2layer(track, color="green", name=""):
    return ipyleaflet.Polyline(
        locations=np.stack([track.lat.values, track.lon.values], axis=1).tolist(), 
        color=color,
        fill=True,
        weight=4,
        name=name
    )

# definition of the base map 
m = ipyleaflet.Map(
    basemap=ipyleaflet.basemaps.Esri.NatGeoWorldMap,
    center=(11.3, -57), zoom=6)
m.add_layer(track2layer(ds_sliced, 'red', 'merian position'))  # adding layer of merian position
m.add_control(ipyleaflet.ScaleControl(position='bottomleft'))
m.add_control(ipyleaflet.LayersControl(position='topright'))
m.add_control(ipyleaflet.FullScreenControl())
display(m)
```

> 
> ```{code-cell} ipython3
> 
> ```

Reviewer: The empty cell at the end can be removed.
```suggestion
```
